pkg/cn_controller/cn_controller.go

```diff
@@ -73,17 +75,30 @@ func (cc *CnController) Sync(ctx context.Context, src *srapi.StarRocksCluster) e
 		return nil
 	}
 
+	//get the cn configMap for resolve ports.
+	//2. get config for generate statefulset and service.
+	config, err := cc.GetCnConfig(ctx, &cnSpec.ConfigMapInfo, src.Namespace)
+	if err != nil {
+		klog.Error("CnController Sync ", "resolve cn configmap failed, namespace ", src.Namespace, " configmapName ", cnSpec.ConfigMapInfo.ConfigMapName, " configMapKey ", cnSpec.ConfigMapInfo.ResolveKey, " error ", err)
+		return err
+	}
+
+	feconfig, _ := cc.getFeConfig(ctx, &src.Spec.StarRocksFeSpec.ConfigMapInfo, src.Namespace)
+	//merge fe config for simplify processing.
+	for k, v := range feconfig {
+		config[k] = v
```

Comment:
it is possible that fe config and cn config has the same config key, this merge operation requires that the two components can't not have same key name, otherwise one will overwrite the other.

Doesn't sound reasonable just to add new restrictions to components configuration because of this operational detailed implementation thing.